.github/workflows/branch.yml

```diff
@@ -0,0 +1,48 @@
+name: "Branch"
+
+on:
+  push:
+
+jobs:
+  e2e:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+    - uses: engineerd/setup-kind@v0.4.0
+      with:
+        version: "v0.9.0"
+        skipClusterCreation: "true"
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v1
+      with:
+        python-version: "3.8"
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade virtualenv
+    - name: Execute E2E tests
+      run: |
+        make test-e2e
+    - name: Upload E2E logs
+      continue-on-error: true
+      uses: actions/upload-artifact@v1
+      with:
+        name: e2elogs
+        path: e2elogs
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v1
+      with:
+        platforms: all
+    - name: Set up Docker Buildx
+      id: buildx
+      uses: docker/setup-buildx-action@v1
+      with:
+        version: latest
+    - name: Available platforms
+      run: echo ${{ steps.buildx.outputs.platforms }}
+    - name: Build all arch images
+      run: |
+        make buildx
+    - name: Publish image
+      run: |
+        docker login ghcr.io -u "${{ secrets.DOCKER_USERNAME }}" -p "${{ secrets.DOCKER_PASSWORD }}"
+        make TAG_VERSION=$GITHUB_SHA publish_container_image_multiarch
```

Comment:
Why we need a new pipeline for branches?

Reply:
Otherwise we cannot test Github Actions when developing a new feature, this would publish an image with SHA of the commit that could be used in real cluster.

I want to make another action that removes all SHA images except the last 3